libr/bin/format/elf/elf.c

```diff
@@ -2076,15 +2258,26 @@ static RBinElfSymbol* get_symbols_from_phdr (struct Elf_(r_bin_elf_obj_t) *bin,
 	if (addr_sym_table > bin->size || addr_sym_table + size > bin->size)
 		goto beach;
 
+	for (i = 0; i < nsym; i++) {
+		r = r_buf_read_at (bin->b, addr_sym_table + i * sizeof (Elf_(Sym)),
+					s, sizeof (Elf_(Sym)));
+		if (r < 1) goto beach;
 #if R_BIN_ELF64
-	r = r_buf_fread_at (bin->b, addr_sym_table, (ut8*)sym,
-			bin->endian ? "I2cS2L": "i2cs2l", nsym);
+		sym[i].st_name = r_read_ble32 (s, bin->endian);
```

Comment:
Yes you can. This is why its called macro and not function

> On 27 Aug 2016, at 04:32, Damien Zammit <notifications@github.com> wrote:
> 
> In libr/bin/format/elf/elf.c:
> 
> >  #if R_BIN_ELF64
> > -	r = r_buf_fread_at (bin->b, addr_sym_table, (ut8*)sym,
> > -			bin->endian ? "I2cS2L": "i2cs2l", nsym);
> > +		sym[i].st_name = r_read_ble32 (s, bin->endian);
> I don't think you can increment i inside a braced initializer...
> …
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>